src/Avalonia.Themes.Fluent/Accents/AccentColors.xaml.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using Avalonia.Markup.Xaml;
+using Avalonia.Media;
+using Avalonia.Styling;
+
+namespace Avalonia.Themes.Fluent.Accents
+{
+    public class AccentColors : Styles
+    {
+        public AccentColors()
+        {
+            AvaloniaXamlLoader.Load(this);
+
+            var accentcolor = GetAccentColor();
+            var light1 = ChangeLuminosity(accentcolor, 0.3);
+            var light2 = ChangeLuminosity(accentcolor, 0.5);
+            var light3 = ChangeLuminosity(accentcolor, 0.7);
+            var dark1 = ChangeLuminosity(accentcolor, -0.3);
+            var dark2 = ChangeLuminosity(accentcolor, -0.5);
+            var dark3 = ChangeLuminosity(accentcolor, -0.7);
+
+
+            this.Resources.Add("SystemAccentColor", accentcolor);
+            this.Resources.Add("SystemAccentColorLight1", light1);
+            this.Resources.Add("SystemAccentColorLight2", light2);
+            this.Resources.Add("SystemAccentColorLight3", light3);
+            this.Resources.Add("SystemAccentColorDark1", dark1);
+            this.Resources.Add("SystemAccentColorDark2", dark2);
+            this.Resources.Add("SystemAccentColorDark3", dark3);
+        }
+
+        private Color GetAccentColor()
+        {
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows); //check if the OS is Windows
+            var defaultValue = "#FF0078D7";// default Avalonia Accent Color
+
+            switch (isWindows)
+            {
+                case true:
+                    return GetAccentColorCore("ImmersiveSystemAccent");
+                case false:
+                    return Color.Parse(defaultValue);
+            }
+        }
+
+        [DllImport("uxtheme.dll", EntryPoint = "#95", CharSet = CharSet.Unicode)]
+        private static extern uint GetImmersiveColorFromColorSetEx(uint dwImmersiveColorSet, uint dwImmersiveColorType, bool bIgnoreHighContrast, uint dwHighContrastCacheMode);
```

Comment:
I don't like platform specific code in Themes project.
Can you create interface with platform-theme specific properties/methods? So it will be easier to implement that interface for another platforms and mock for tests.

Reply:
You can register that interface/implementation here:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Windows/Avalonia.Win32/Win32Platform.cs#L118

And later get implementation by interface from static AvaloniaLocator.

Reply:
now I'm working in that, also I'll put the unmanaged code in Avalonia.Visuals (Avalonia.Win32 depends of this)